GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1175,9 +1175,11 @@ private void UpdatePlaceholders()
                             // 1. not in the projection anymore
                             // 2. in the projection but is not a folder in the projection
                             // 3. Folder placeholder is a possible tombstone
-                            if (!isProjected ||
+                            // and if the folder is not the root folder of the repository
+                            if ((!isProjected ||
                                 !isFolder ||
-                                folderPlaceholder.IsPossibleTombstoneFolder)
+                                folderPlaceholder.IsPossibleTombstoneFolder) &&
+                                folderPlaceholder.Path != string.Empty)

```

Comment:
We prefer `!string.IsNullOrEmpty(folderPlaceholder.Path)` for comparing against an empty string.

Reply:
Thanks for the tip -- while I didn't use it here because of @kewillford's suggestion below, I did end up using it in #1128!